src/iptvsimple/utilities/StreamUtils.cpp

```diff
@@ -63,7 +63,37 @@ void StreamUtils::SetAllStreamProperties(PVR_NAMED_VALUE* properties, unsigned i
     }
     else // inputstream.adpative
     {
-      StreamUtils::SetStreamProperty(properties, propertiesCount, propertiesMax, PVR_STREAM_PROPERTY_STREAMURL, streamURL);
+      // If no media headers are explicitly set for inputstream.adaptive,
+      // strip the headers from streamURL and put it to media headers property
+
+      bool haveStreamHeaders = !channel.GetProperty("inputstream.adaptive.stream_headers").empty();
+      if (!haveStreamHeaders)
+      {
+        // preserve any kodi protocol options after "|"
+        std::string url = streamURL;
```

Comment:
Defining these here serves no purpose. They can be defined where are used. Also they should be const.